pilot/pkg/features/experimental.go

```diff
@@ -0,0 +1,249 @@
+		"The version to be used for the Kubernetes Multi-Cluster Services (MCS) API.").Get()
+
+	EnableMCSAutoExport = env.Register(
+		"ENABLE_MCS_AUTO_EXPORT",
+		false,
+		"If enabled, istiod will automatically generate Kubernetes "+
+			"Multi-Cluster Services (MCS) ServiceExport resources for every "+
+			"service in the mesh. Services defined to be cluster-local in "+
+			"MeshConfig are excluded.",
+	).Get()
+
+	EnableMCSServiceDiscovery = env.Register(
+		"ENABLE_MCS_SERVICE_DISCOVERY",
+		false,
+		"If enabled, istiod will enable Kubernetes Multi-Cluster "+
+			"Services (MCS) service discovery mode. In this mode, service "+
+			"endpoints in a cluster will only be discoverable within the "+
+			"same cluster unless explicitly exported via ServiceExport.").Get()
+
+	EnableMCSHost = env.Register(
+		"ENABLE_MCS_HOST",
+		false,
+		"If enabled, istiod will configure a Kubernetes Multi-Cluster "+
+			"Services (MCS) host (<svc>.<namespace>.svc.clusterset.local) "+
+			"for each service exported (via ServiceExport) in at least one "+
+			"cluster. Clients must, however, be able to successfully lookup "+
+			"these DNS hosts. That means that either Istio DNS interception "+
+			"must be enabled or an MCS controller must be used. Requires "+
+			"that ENABLE_MCS_SERVICE_DISCOVERY also be enabled.").Get() &&
+		EnableMCSServiceDiscovery
+
+	EnableMCSClusterLocal = env.Register(
+		"ENABLE_MCS_CLUSTER_LOCAL",
+		false,
+		"If enabled, istiod will treat the host "+
+			"`<svc>.<namespace>.svc.cluster.local` as defined by the "+
+			"Kubernetes Multi-Cluster Services (MCS) spec. In this mode, "+
+			"requests to `cluster.local` will be routed to only those "+
+			"endpoints residing within the same cluster as the client. "+
+			"Requires that both ENABLE_MCS_SERVICE_DISCOVERY and "+
+			"ENABLE_MCS_HOST also be enabled.").Get() &&
+		EnableMCSHost
+
+	EnableAnalysis = env.Register(
+		"PILOT_ENABLE_ANALYSIS",
+		false,
+		"If enabled, pilot will run istio analyzers and write analysis errors to the Status field of any "+
+			"Istio Resources",
+	).Get()
+
+	AnalysisInterval = func() time.Duration {
+		val, _ := env.Register(
+			"PILOT_ANALYSIS_INTERVAL",
+			10*time.Second,
+			"If analysis is enabled, pilot will run istio analyzers using this value as interval in seconds "+
+				"Istio Resources",
+		).Lookup()
+		if val < 1*time.Second {
+			log.Warnf("PILOT_ANALYSIS_INTERVAL %s is too small, it will be set to default 10 seconds", val.String())
+			return 10 * time.Second
+		}
+		return val
+	}()
+
+	EnableStatus = env.Register(
+		"PILOT_ENABLE_STATUS",
+		false,
+		"If enabled, pilot will update the CRD Status field of all istio resources with reconciliation status.",
+	).Get()
+
+	StatusUpdateInterval = env.Register(
+		"PILOT_STATUS_UPDATE_INTERVAL",
+		500*time.Millisecond,
+		"Interval to update the XDS distribution status.",
+	).Get()
+
+	StatusQPS = env.Register(
+		"PILOT_STATUS_QPS",
```

Comment:
tuning? But then PILOT_ENABLE_STATUS is not and its logical to group. Tricky to pick just one category!

Reply:
move these to tuning